extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php

```diff
@@ -188,11 +195,44 @@ private function getThemeName( $params, $value ) {
 
 	private function getLayout( $params ) {
 		$layoutName = isset( $params[ 'layout' ] ) ? $params[ 'layout' ] : false;
-		if ( $layoutName && in_array( $layoutName, $this->supportedLayouts ) ) {
+		if ( $this->getParamsValidator()->validateLayout( $layoutName ) ) {
 			//make sure no whitespaces, prevents side effects
 			return self::INFOBOX_LAYOUT_PREFIX . $layoutName;
 		}
 
 		return self::INFOBOX_LAYOUT_PREFIX . self::DEFAULT_LAYOUT_NAME;
 	}
+
+	private function getColor( $colorParam, $params, PPFrame $frame ) {
+		$sourceParam = $colorParam . '-source';
+		$defaultParam = $colorParam . '-default';
+
+		$color = '';
```

Comment:
I'm not sure about returning empty string as a default color value when color is invalid or not present. What do you think about using null?